src/test/Spec.hs

```diff
@@ -193,19 +199,23 @@ integrationTests = testGroup "Solidity Integration Testing"
   ]
 
 testContract :: FilePath -> Maybe FilePath -> [(String, Campaign -> Bool)] -> TestTree
-testContract fp cfg as = testCase fp $ do
+testContract = flip testContract' Nothing
+
+testContract' :: FilePath -> Maybe Text -> Maybe FilePath -> [(String, Campaign -> Bool)] -> TestTree
+testContract' fp n cfg as = testCase fp $ do
   c <- set (sConf . quiet) True <$> maybe (pure defaultConfig) (fmap _econfig . parseConfig) cfg
-  res <- runContract fp c
+  res <- runContract fp n c
   mapM_ (\(t,f) -> assertBool t $ f res) as
 
-runContract :: FilePath -> EConfig -> IO Campaign
-runContract fp c =
+runContract :: FilePath -> Maybe Text -> EConfig -> IO Campaign
+runContract fp n c =
   flip runReaderT c $ do
     g <- getRandom
-    (v,w,ts) <- loadSolTests fp Nothing
-    cs  <- contracts fp
-    ads <- addresses
-    campaign (pure ()) v w ts (Just $ mkGenDict 0.15 (extractConstants cs ++ NE.toList ads) [] g (returnTypes cs))
+    (v,w,ts) <- loadSolTests fp n
+    cs  <- Echidna.Solidity.contracts fp
+    ads <- NE.toList <$> addresses
+    let ads' = AbiAddress . addressWord160 <$> v ^. env . EVM.contracts . to keys
```

Comment:
It seems this is used in several places of the code. Maybe we can have a function for this?

Reply:
let's sort this out after this PR